multiversx_sdk/validators/validators_transactions_factory.py

```
@@ -0,0 +1,280 @@
+
+        num_of_nodes = validators_file.get_num_of_nodes()
+
+        call_arguments: list[ISingleValue] = []
+        call_arguments.append(U32Value(num_of_nodes))
+
+        validator_signers = validators_file.get_signers()
+
+        for validator in validator_signers:
+            signed_message = validator.sign(node_operator.get_public_key())
+
+            call_arguments.append(BytesValue(validator.secret_key.generate_public_key().buffer))
+            call_arguments.append(BytesValue(signed_message))
+
+        if rewards_address:
+            call_arguments.append(AddressValue.new_from_address(rewards_address))
+
+        args = self.serializer.serialize_to_parts(call_arguments)
+        return data_parts + [arg.hex() for arg in args]
+
+    def create_transaction_for_topping_up(
+        self,
+        sender: Address,
+        amount: int,
+    ) -> Transaction:
+        data = ["stake"]
+
+        transaction = TransactionBuilder(
+            config=self.config,
+            sender=sender,
+            receiver=Address.new_from_hex(STAKING_SMART_CONTRACT_ADDRESS_HEX),
+            data_parts=data,
+            gas_limit=self.config.gas_limit_for_topping_up,
+            add_data_movement_gas=True,
+            amount=amount,
+        ).build()
+
+        return transaction
+
+    def create_transaction_for_unstaking(
+        self,
+        sender: Address,
+        public_keys: list[ValidatorPublicKey],
+    ) -> Transaction:
+        data_parts = ["unStake"] + [key.hex() for key in public_keys]
+
+        transaction = TransactionBuilder(
+            config=self.config,
+            sender=sender,
+            receiver=Address.new_from_hex(STAKING_SMART_CONTRACT_ADDRESS_HEX),
+            data_parts=data_parts,
+            gas_limit=self.config.gas_limit_for_unstaking * len(public_keys),
+            add_data_movement_gas=True,
+        ).build()
+
+        return transaction
+
+    def create_transaction_for_unjailing(
```

maybe throw an error if the amount isn't equal to 2.5 egls * num_keys ?